### ISSUE TITLE
Ability to add comments in an expense

### ISSUE THREAD

**Issue by vishwa5854** [oss-apps/split-pro#61]
Whenever some one wants to leave from an expense, they could mention why they want to be excluded from an expense and they could leave that particular expense, I have extensively used this feature on splitwise, and would like to see it here.

**Comment by KMKoushik** [comment]
@vishwa5854 Thank you for the feedback. We already have the DB support for comments. Would you be interested in working on building the UI for it?

https://github.com/oss-apps/split-pro/blob/main/prisma/schema.prisma#L173

**Comment by vishwa5854** [comment]
Sure, I will work on it!

**Comment by KMKoushik** [comment]
@vishwa5854  Thank you so much. Let me know if you need any help on setting up the project. This is relatively new project and I'm trying to improve the docs

**Comment by Himanshu9271** [comment]
hi, is someone working on it ? if not i would like to give it a try.

**Comment by vishwa5854** [comment]
Hey @Himanshu9271 , I am not working on it, you can go ahead

**Comment by Himanshu9271** [comment]
sure, I'll start working on it.

**Comment by FelixDz** [comment]
@Himanshu9271 did you make some progress on this? This feature is quite important for my group use case, so we would be happy to see it happening. I'm not sure about my skills with this dev environment but if you need some help, let us know!

**Comment by Himanshu9271** [comment]
Hi @FelixDz. This usecase is also important for my group as well. 
I was able to get split-pro running on a temporary workstation to understand the codebase, but when i tried to set it up on my personal PC few days later to start working on it I got the following error:
`Error: Error: P1013: The provided database string is invalid. invalid port number in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.`
I verified the `.env` file and  tried percentage-encoding the database url parameters, but still geting the same error.  
Update:
I was able to work around the issue by writing the final, concatenated connection string directly into the DATABASE_URL variable like `DATABASE_URL="postgresql://splitpro:password@localhost:5432/splitpro?schema=public"`
Will work on the feature now.

**Comment by krokosik** [comment]
Yes, my bad on that one. Turns out prisma does not interpolate env vars the way docker does :/

**Comment by krokosik** [comment]
Let me know if you need any help with the feature :)

**Comment by Himanshu9271** [comment]
Hi @krokosik 
I have forked the repo and made code changes to feature/expense-comments branch. Sorry for asking if it is obvious, but how do i add unit test ? I have created one trpc api to add comment and added Expense Comment compoenent. I am asking this because this was one of the prerequisite for the PR.

**Comment by krokosik** [comment]
You can submit the PR and we can think about adding unit tests then :)
It will also be easier for me to answer as unit tests don't always makes sense and this case seems like integration tests would be more applicable (which we don't have as of now)

**Comment by Himanshu9271** [comment]
Submitted the PR.

**Comment by krokosik** [comment]
Nice, will take a look at the PR once I finish the current release

**Comment by ConfidentKeyboard** [comment]
Any updates on this feature?

**Comment by krokosik** [comment]
@ConfidentKeyboard it was squeezed out to 2.2 due to more pressing issues and I want to create a release before going on holidays. It is something I would like to work on next
